lib/pbench/test/unit/server/test_indexing_tarballs.py

```diff
@@ -272,6 +273,35 @@ def find_dataset(self, resource_id: str):
         )
 
 
+class FakeAudit:
+    BACKGROUND_USER = "test"
+    audits: list["FakeAudit"] = []
+    sequence: int = 1
+
+    def __init__(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        if not hasattr(self, "id"):
+            self.id = self.sequence
+            __class__.sequence += 1
```

Comment:
If it _does_ have an `id` attribute, shouldn't we be setting `__class__.sequence = self.id + 1`?  E.g.,

```
        if not hasattr(self, "id"):
            self.id = __class__.sequence
        __class__.sequence = self.id + 1
```
Otherwise, I think we risk producing multiple records with the same ID.

Reply:
We don't (and rarely would) construct a DB record with a pre-assigned `id` value; but if, for any reason, we choose to do so in testing (e.g., to validate duplicate behavior), we want to use that number. The normal case is that `id` isn't assigned manually, and we want to assign it sequentially as the DB engine would.

Reply:
OK, so creating an instance with an explicitly-specified ID is not supposed to have any effect on the ID sequencing for normally-created records...and, it's just supposed to be understood that specifying an ID is going to create a potentially-duplicate ID (whether or not it's an actual duplicate depends on the specified value and what other records are created).